experiments/common/initialize_sinara.py

```diff
@@ -0,0 +1,122 @@
+
+    @host_only
+    def get_dds_and_ttls(self):
+        """Gets DDS and TTL objects that are initialized and uninitialized.
+
+        Populates self.initialized_ddses, self.uninitialized_ddses,
+        self.initialized_ttls, and self.uninitialized_ttls. The first element
+        in each of these lists is a placeholder element for determining element type.
+        """
+        dds_params = pyon.decode(self.cxn.artiq.get_dds_parameters())
+        dummy_dds = self.get_device(self.devices.ad9910s[0])
+        self.initialized_ddses = [(dummy_dds, [0., 0., 0., 0., 0.])]  # DDSes in artiq server.
+        self.uninitialized_ddses = [(dummy_dds, "")]  # DDSes not in artiq server.
+        for name in dds_params:
+            if name in self.devices.ad9910s:
+                self.initialized_ddses.append((self.get_device(name), dds_params[name]))
+            else:
+                self.cxn.artiq.remove_sinara_dds(name)
+        for name in self.devices.ad9910s:
+            if name not in dds_params:
+                self.uninitialized_ddses.append((self.get_device(name), name))
+
+        ttl_params = pyon.decode(self.cxn.artiq.get_ttl_parameters())
+        dummy_ttl = self.get_device(self.devices.ttl_outs[0])
+        self.initialized_ttls = [(dummy_ttl, 0.)]
+        self.uninitialized_ttls = [(dummy_ttl, "")]
+        for name in ttl_params:
+            if name in self.devices.ttl_outs:
+                self.initialized_ttls.append((self.get_device(name), ttl_params[name]))
+            else:
+                self.cxn.artiq.remove_sinara_ttl(name)
+        for name in self.devices.ttl_outs:
+            if name not in ttl_params:
+                self.uninitialized_ttls.append((self.get_device(name), name))
+
+    @kernel
+    def run_kernel(self):
+        self.core.reset()
+        for kk in range(len(self.initialized_ddses)):
+            if kk > 0:
+                dds, values = self.initialized_ddses[kk]
+                self._set_dds(dds, values)
+
+        for kk in range(len(self.uninitialized_ddses)):
+            if kk > 0:
+                dds, name = self.uninitialized_ddses[kk]
+                self.get_dds(dds, name)
+
+        for kk in range(len(self.initialized_ttls)):
+            if kk > 0:
+                ttl, value = self.initialized_ttls[kk]
+                self._set_ttl(ttl, value)
+
+        for kk in range(len(self.uninitialized_ttls)):
+            if kk > 0:
+                ttl, name = self.uninitialized_ttls[kk]
+                self.get_ttl(ttl, name)
+
+    @kernel
+    def get_dds(self, device, name):
+        """Get the DDS values from the sinara hardware.
+
+        It gets the frequency, phase, amplitude, and attenuation from the DDS,
+        and assumes that the DDS state is -1. (off).
+        """
+        self.core.break_realtime()
+        frequency, phase, amplitude = device.get()
+        self.core.break_realtime()
+        attenuation = device.get_att()
+        state = -1.
```

Comment:
for our experiments is it more efficient to assume that the DDS state is off/if it's on will this create errors elsewhere?